### PR TITLE
Fixing global variable bug

### PR DESCRIPTION
**Pull request by Rutrus**
Now you can invite by id or by name (in contact_list)

### CODE REVIEW

**Comment by Walkman100** [comment]
Is there a difference between a variable being called `user_` and `user`? if not, I see no changes that would fix the bug...

File: plugins/invite.lua [hunk 1]
```diff
@@ -7,41 +7,45 @@ do
...
-  description = "Invite other user to the chat group", 
+  description = "Invite other user to the chat group",
   usage = {
     "!invite name [user_name]", 
     "!invite id [user_id]" },
   patterns = {
-    "^!invite (name) (.*)$",
-    "^!invite (id) (%d+)$"
+    "^!invite (name) (.*)",
+    "^!invite (id) (%d+)"
```
Comment (Walkman100) [inline plugins/invite.lua]: Oh wait, is this the bug? If so, ignore my comment above

Comment (rockneurotiko) [inline plugins/invite.lua]: No, the $ at the end is important, if not the pattern match can get lines that shouldn't
For example, the second one without the dollar match this command, and shouldn't:
`!invite id 123123123 hi :)`

**Comment by rockneurotiko** [comment]
I don't know what "global variable error" you are "fixing".
The plugin right now let you invite by id or name, and you are who are inserting global variables (that you shouldn't)
Can you explaing what error is?

File: plugins/invite.lua [hunk 2]
```diff
@@ -7,41 +7,45 @@ do
...
-  end
-  
+    user_ = matches[2]
+    user_ = string.gsub(user_," ","_")  
   -- User submitted an id
-  if matches[1] == "id" then
-    local user = matches[2]
-    user = 'user#id'..user
+  elseif matches[1] == "id" then
+    user_ = matches[2]
```
Comment (yagop) [inline plugins/invite.lua]: Don't declare global variables

File: bot/bot.lua [hunk 3]
```diff
@@ -41,48 +41,41 @@ function on_binlog_replay_end()
...
-  if msg.out then
+  if msg.out then -- Dont process outgoing messages
     print('\27[36mNot valid: msg from us\27[39m')
     return false
-  end
-
-  -- Before bot was started
-  if msg.date < now then
+    
+  elseif msg.date < now then -- When just started
```
Comment (yagop) [inline bot/bot.lua]: `elseif ` after return is the same as `else` doesn't need to be changed

**Comment by yagop** [comment]
The name of the PR seems wrong and there should be 2 PR.

**Pull request closed without merging**